flink-java/src/main/java/org/apache/flink/api/java/utils/ParameterTool.java

```diff
@@ -548,6 +565,7 @@ protected Object clone() throws CloneNotSupportedException {
 	public ParameterTool mergeWith(ParameterTool other) {
 		ParameterTool ret = new ParameterTool(this.data);
 		ret.data.putAll(other.data);
+		ret.unrequestedParameters.addAll(other.data.keySet());
```

Comment:
The docs should probably state how unrequested parameters are handles here, i.e. the returned `ParameterTool` will have have its unrequested parameters reset. An alternative idiom may be to actually keep the old `unrequestedParameters` in `ret` and merge them with `other.unrequestedParameters`. What do you thinks makes most sense?